src/bundler/bundle_v2.zig

```diff
@@ -943,33 +954,52 @@ pub const BundleV2 = struct {
-                } else {}
-            }
+            try this.graph.entry_points.ensureUnusedCapacity(this.graph.allocator, entry_points.len);
+            try this.graph.input_files.ensureUnusedCapacity(this.graph.allocator, entry_points.len);
+            try this.graph.path_to_source_index_map.ensureUnusedCapacity(this.graph.allocator, @as(u32, @truncate(entry_points.len)));
 
-            for (bake_entry_points) |entry_point| {
-                const resolved = this.bundler.resolveEntryPoint(entry_point.path) catch continue;
-                if (try this.enqueueItem(null, &batch, resolved, true, switch (entry_point.graph) {
-                    .client => .browser,
-                    .server => this.bundler.options.target,
-                    .ssr => .kit_server_components_ssr,
-                })) |source_index| {
-                    this.graph.entry_points.append(this.graph.allocator, Index.source(source_index)) catch unreachable;
-                } else {}
+            for (entry_points) |entry_point| {
+                switch (variant) {
+                    .normal => {
+                        const resolved = this.bundler.resolveEntryPoint(entry_point) catch
+                            continue;
+                        const source_index = try this.enqueueItem(null, &batch, resolved, true, this.bundler.options.target) orelse
+                            continue;
+                        try this.graph.entry_points.append(this.graph.allocator, Index.source(source_index));
+                    },
+                    .dev_server => {
+                        // Dev server provides target and some extra integration.
+                        const resolved = this.bundler.resolveEntryPoint(entry_point.path) catch
+                            continue;
+                        const source_index = try this.enqueueItem(null, &batch, resolved, true, switch (entry_point.graph) {
+                            .client => .browser,
+                            .server => this.bundler.options.target,
+                            .ssr => .kit_server_components_ssr,
+                        }) orelse continue;
 
-                if (entry_point.route_index.unwrap()) |route_index| {
-                    _ = try this.bundler.options.dev_server.?.server_graph.insertStaleExtra(resolved.path_pair.primary.text, false, true, route_index);
+                        try this.graph.entry_points.append(this.graph.allocator, Index.source(source_index));
+
+                        if (entry_point.route_index.unwrap()) |route_index| {
+                            _ = try this.bundler.options.dev_server.?.server_graph.insertStaleExtra(resolved.path_pair.primary.text, false, true, route_index);
+                        }
+
+                        if (entry_point.css) {
+                            try data.css_data.putNoClobber(this.graph.allocator, Index.init(source_index), .{ .imported_on_server = false });
+                        }
```

Comment:
Should this block execute when `enqueueItem` returns null?

Reply:
it shouldnt happen. a better thing to have would be `orelse @panic("")` continue is fine.